src/embeddings/embeddings_calculator.cc

```diff
@@ -116,8 +116,20 @@ class EmbeddingsCalculator : public CalculatorBase {
         ::InferenceOutput embeddingsOutputMap;
         ::InferenceInput embeddingsInputMap;
         size_t received_batch_size = 1;
-        size_t max_context_length = 512;  // default allowed input length. Otherwise, it will be read from model rt_info>config>max_position_embeddings in the model.xml file
+        size_t max_context_length = 1024;  // default allowed input length. Otherwise, it will be read from model rt_info>config>max_position_embeddings in the model.xml file
         ov::AnyMap modelConfig = embeddings_session->getModelConfig();
+        try {
+            if (modelConfig.count("max_position_embeddings")) {
+                max_context_length = modelConfig["max_position_embeddings"].as<size_t>();
+            } else if (modelConfig.count("max_trained_positions")) {
+                max_context_length = modelConfig["max_trained_positions"].as<size_t>();
+            } else {
+                SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "model_info->max_position_embeddings nor max_trained_positions included in model rt_info. Using default value {}", max_context_length);
```

Comment:
Is this message understandable for the user? `model_info->max_position_embeddings` looks more like something from the internals not the interface.

Reply:
it is not the c++ code but rt_info structure in the model.xml